blockio/src-linux/System/FS/BlockIO/Internal/Fcntl.hsc

```diff
@@ -0,0 +1,27 @@
+{-# LANGUAGE CPP #-}
+module System.FS.BlockIO.Internal.Fcntl (fileSetCaching) where
+
+#if MIN_VERSION_unix(2,8,7)
+
+import System.Posix.Fcntl (fileSetCaching)
+
+#else
+
+#define _GNU_SOURCE
```

Comment:
Through other channels you explained to me why we need this `#define` here. For posterity, maybe we should link to documentation that explains why this has to be included?